### PR TITLE
Create PerfettoEventProcessor for processing trace events

### PR DESCRIPTION
**Pull request by kenzieschmoll**
This PR takes the existing timeline_event_processor file and breaks it up into two classes: `BaseTraceEventProcessor` and `LegacyTraceEventProcessor`. It also creates a class `PerfettoTraceEventProcessor` for when the perfetto trace viewer is in use. `PerfettoTraceEventProcessor` and `LegacyTraceEventProcessor` both extend `BaseTraceEventProcessor`.

### CODE REVIEW

File: packages/devtools_app/lib/src/screens/performance/panes/timeline_events/legacy/legacy_event_processor.dart [hunk 1]
```diff
@@ -0,0 +1,233 @@
+// Copyright 2020 The Chromium Authors. All rights reserved.
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/screens/performance/panes/timeline_events/legacy/legacy_event_processor.dart:1]: the code in this file was moved from timeline_event_processor.dart (reviewers - you may want to review there first to see what you're looking for in this file).

File: packages/devtools_app/lib/src/screens/performance/panes/timeline_events/perfetto/perfetto_event_processor.dart [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+      final frameNumberFromEvent = eventWrapper.flutterFrameNumber;
+      if (frameNumberFromEvent != null) {
+        _startFrameId ??= frameNumberFromEvent;
+        // We process events in timestamp order, so [_endFrameId] will always be
+        // reassigned.
+        _endFrameId = max(_endFrameId ?? -1, frameNumberFromEvent);
+      }
+      if (frameNumberFromEvent == 1018) {
+        print('this is the event ');
+      }
```
Comment (polina-c) [inline packages/devtools_app/lib/src/screens/performance/panes/timeline_events/perfetto/perfetto_event_processor.dart:63]: Is it forgotten debug output?
If no, can you put 1018 into constant?

**Review by polina-c — APPROVED** [review]
lgtm with some comments

File: packages/devtools_app/lib/src/screens/performance/panes/timeline_events/perfetto/perfetto_event_processor.dart (hunk 2 above)
Comment (kenzieschmoll) [reply packages/devtools_app/lib/src/screens/performance/panes/timeline_events/perfetto/perfetto_event_processor.dart:63]: removed 👍

**Pull request merged**